### ISSUE TITLE
[Classic] Pets attack melee bug + following speed bug!

### ISSUE THREAD

**Issue by cooler-SAI** [cmangos/issues#1147]
**Current behavior**: 
Pets attack melee bug + following speed bug!
**Expected behavior**: 
When Imp loose mana he must attack melee, then use fireballs again - but he don't/ And they dissapear when following player 
**Steps to reproduce**:

1. Create Warlock and Summon Imp
2. Command attack target to Imp
3. See what happened when he loose Mana. Also Summon Pet or Imp, go forward and you will see that soon Pet dissapear ( loose speed and dissapear)

**Client version**: 1.12.1

**Commit hash**: https://github.com/cmangos/mangos-classic/commit/a82feff923eac73efb07c62c0f4e9a5df4022dc4

**Database version**: Last Classic

**Operating system**: Windows 7 x64

Also summoning @Phatcat for this issue))))
[//]: # (If this is a crash report, include the crashlog from a debug build with https://gist.github.com/)

**Comment by Phatcat** [comment]
> When Imp loose mana he must attack melee

No. Imps were not and never should be, able to melee. At least not the warlock player's imp. It's the same for the Mage's Water Elemental. That one ain't supposed to be able to melee either.

> Also Summon Pet or Imp, go forward and you will see that soon Pet dissapear ( loose speed and dissapear)

Please elaborate on this one, how exactly did the pet disappear and what exactly do you mean by 'lose speed'?

**Comment by Phatcat** [comment]
Dunno why I closed this, musy have pressed the wrong buton when commenting. Weird I didn't notice it, I don't remember closing this at all..

Well, first part can be overlooked though since it's intended as I explained, and the second part we are already aware of, the entire pet following system needs some polishing, including their movement speed out of combat as well. More research needs to be done, though. We're getting there.
